### PR TITLE
[DOCS] Adds cohere service to inference API docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR expands the PUT inference API docs with the `cohere` service parameters and example.

### Preview

[PUT inference API docs](https://elasticsearch_bk_105394.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-inference-api.html)

### CODE REVIEW

File: docs/reference/inference/put-inference.asciidoc [hunk 1]
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
 (Required, string)
 The type of service supported for the specified task type.
 Available services:
+* `cohere`:
```
Comment (jonathan-buttner) [inline docs/reference/inference/put-inference.asciidoc:68]: How about we add:

```
specify the `text_embedding` task type to use the Cohere service.
```

File: docs/reference/inference/put-inference.asciidoc [hunk 2]
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
...
+https://dashboard.cohere.com/api-keys[on the API keys settings page].
+
+IMPORTANT: You need to provide the API key only once, during the {infer} model 
+creation. The <<get-inference-api>> does not retrieve your API key. After 
+creating the {infer} model, you cannot change the associated API key. If you 
+want to use a different API key, delete the {infer} model and recreate it with 
+the same name and the updated API key.
+
+`embedding_type`::
+(Required, string)
```
Comment (jonathan-buttner) [inline docs/reference/inference/put-inference.asciidoc:95]: Let's switch this to `Optional`. We can say it defaults to `float` if not specified: https://docs.cohere.com/reference/embed under `embedding_types`

File: docs/reference/inference/put-inference.asciidoc [hunk 3]
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
...
+Specifies the types of embeddings you want to get back.
+Valid values are:
+  * `float`: use it for the default float embeddings.
+  * `int8`: use it for signed int8 embeddings.
+
+`model_id`::
```
Comment (jonathan-buttner) [inline docs/reference/inference/put-inference.asciidoc:101]: 👍

File: docs/reference/inference/put-inference.asciidoc [hunk 4]
```diff
@@ -141,11 +173,31 @@ Settings to configure the {infer} task. These settings are specific to the
 .`task_settings` for `text_embedding`
 [%collapsible%closed]
 =====
+`input_type`:::
+(optional, string)
+For `cohere` service only. Specifies the type of input passed to the model.
+Valid values are:
+  * `ingest`: use it for storing document embeddings in a vector database.
+  * `search`: use it for stroing embeddings of search queries run against a
```
Comment (jonathan-buttner) [inline docs/reference/inference/put-inference.asciidoc:181]:
```suggestion
  * `search`: use it for storing embeddings of search queries run against a 
```

File: docs/reference/inference/put-inference.asciidoc [hunk 5]
```diff
@@ -141,11 +173,31 @@ Settings to configure the {infer} task. These settings are specific to the
 .`task_settings` for `text_embedding`
 [%collapsible%closed]
 =====
+`input_type`:::
+(optional, string)
+For `cohere` service only. Specifies the type of input passed to the model.
+Valid values are:
```
Comment (jonathan-buttner) [inline docs/reference/inference/put-inference.asciidoc:179]: I also added

- `"classification"`: Used for embeddings passed through a text classifier.
- `"clustering"`: Used for the embeddings run through a clustering algorithm.

**Review by jonathan-buttner — COMMENTED** [review]
Looks good. Just an idea but as we support more 3rd party services I wonder if it'll make sense to create like a section for each so users can go directly to that section instead of having to look at each field to determine if it is applicable for the service they're trying to configure.

Something like a heading for each service and within the its section the specific fields to configure that servce.

**Comment by szabosteve** [comment]
@jonathan-buttner Thank you for the review. I addressed your feedback via [ead1be8](https://github.com/elastic/elasticsearch/pull/105394/commits/ead1be8482fcb051d1a4e59a1326b82fe47bd940). Restructuring the page by service type is a great idea. I'll think about the best way to do it and cover it in a separate PR if it's okay. I need to open another one for the built-in and Eland models anyway.

**Review by jonathan-buttner — APPROVED** [review]
Thanks for the changes!

**Pull request merged**